### PR TITLE
Create NgrokReverseProxyOnNetwork

### PR DESCRIPTION
**Pull request by praveenthepro**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Solutions/DNS Essentials/NgrokReverseProxyOnNetwork [hunk 1]
```diff
@@ -0,0 +1,30 @@
+id: 50b0dfb7-2c94-4eaf-a332-a5936d78c263
```
Comment (vakohl) [inline Solutions/DNS Essentials/NgrokReverseProxyOnNetwork:1]: @praveenthepro can you place this file under under following path?
https://github.com/Azure/Azure-Sentinel/tree/1dee9c27f3bc886fe690fc15d49e9b7055b462e7/Solutions/DNS%20Essentials/Analytic%20Rules

File: Solutions/DNS Essentials/NgrokReverseProxyOnNetwork [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+name: Ngrok Reverse Proxy on Network (ASIM DNS Solution)
```
Comment (vakohl) [inline Solutions/DNS Essentials/NgrokReverseProxyOnNetwork:2]: I think we are missing few properties like queryFrequency, queryPeriod etc.
Please refer to this template for updating required properties

File: Solutions/DNS Essentials/NgrokReverseProxyOnNetwork [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+description: |
+  'This detection identifies the top 4 Ngrok domains from DNS resolution. Ngrok’s reverse proxy can
+tags:
+  - Schema: ASimDns
+    SchemaVersion: 0.1.6
+requiredDataConnectors: []
+tactics:
+  - Command And Control
```
Comment (vakohl) [inline Solutions/DNS Essentials/NgrokReverseProxyOnNetwork:10]: I don't think space is needed here.
image

File: Solutions/DNS Essentials/NgrokReverseProxyOnNetwork [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  - T1572
+  - T1090
+  - T1102
+query: |
+  // List of Ngrok-like domains
+  let NgrokDomains = dynamic(["ngrok.com", "ngrok.io", "ngrok", "tunnel.com", "korgn", "lennut.com"]);
+  _Im_Dns
+  | where TimeGenerated > ago(1h) // Logs from last hour
+  | where isnotempty(DnsQuery) // Non-empty DNS queries
+  | where DnsQuery has_any (NgrokDomains) // Queries containing Ngrok domains
```
Comment (vakohl) [inline Solutions/DNS Essentials/NgrokReverseProxyOnNetwork:21]: I would do summarize count() by DnsQuery. just to limit duplicate rows with same entry

Comment (praveenthepro) [reply Solutions/DNS Essentials/NgrokReverseProxyOnNetwork:21]: Each event has an separate id in dns events

File: Solutions/DNS Essentials/NgrokReverseProxyOnNetwork [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+tactics:
+  - Command And Control
+relevantTechniques:
+  - T1572
+  - T1090
+  - T1102
+query: |
+  // List of Ngrok-like domains
+  let NgrokDomains = dynamic(["ngrok.com", "ngrok.io", "ngrok", "tunnel.com", "korgn", "lennut.com"]);
+  _Im_Dns
```
[the remark this answers is not in the document]
Comment (praveenthepro) [reply Solutions/DNS Essentials/NgrokReverseProxyOnNetwork:18]: Thanks for letting me know, I will modify the code

[the remark this answers is not in the document]
Comment (praveenthepro) [reply Solutions/DNS Essentials/NgrokReverseProxyOnNetwork]: modified the code

File: Solutions/DNS Essentials/NgrokReverseProxyOnNetwork (hunk 1 above)
Comment (vakohl) [reply Solutions/DNS Essentials/NgrokReverseProxyOnNetwork:1]: @praveenthepro can you please validate the rule have been moved under Analytics Rules? I don't see this change

Comment (vakohl) [reply Solutions/DNS Essentials/NgrokReverseProxyOnNetwork:1]: image

Comment (praveenthepro) [reply Solutions/DNS Essentials/NgrokReverseProxyOnNetwork]: Thanks for noticing, Now changed the path, Please checkout

**Review by rahul0216 — DISMISSED** [review]
Changes look good.

**Review by vakohl — APPROVED** [review]

**Pull request merged**